platform/tests/Platform.ApiTests/Steps/EstablishmentLocalAuthoritiesSteps.cs

```diff
@@ -43,60 +42,131 @@ private void GivenAValidLocalAuthorityRequestWithId(string id)
-    private void ThenTheLocalAuthoritiesSuggestResultShouldBeOk()
+    [Then("the local authority result should be correct")]
+    private async Task ThenTheLocalAuthorityResultShouldBeCorrect()
     {
-        var result = _api[SuggestRequestKey].Response;
+        var response = _api[RequestKey].Response;
 
-        result.Should().NotBeNull();
-        result.StatusCode.Should().Be(HttpStatusCode.OK);
+        response.Should().NotBeNull();
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var content = await response.Content.ReadAsByteArrayAsync();
+        var result = content.FromJson<LocalAuthority>();
+
+        result.Code.Should().Be("201");
+        result.Name.Should().Be("City of London");
     }
 
-    [Then("the local authorities search result should be ok")]
-    private void ThenTheLocalAuthoritiesSearchResultShouldBeOk()
+    [Then("the local authority result should be not found")]
+    private void ThenTheLocalAuthorityResultShouldBeNotFound()
     {
-        var result = _api[SearchRequestKey].Response;
+        var result = _api[RequestKey].Response;
 
         result.Should().NotBeNull();
-        result.StatusCode.Should().Be(HttpStatusCode.OK);
+        result.StatusCode.Should().Be(HttpStatusCode.NotFound);
     }
 
-    [Then("the local authorities query result should be ok")]
-    private void ThenTheLocalAuthoritiesQueryResultShouldBeOk()
+    [Then("the local authorities suggest result should be correct")]
+    private async Task ThenTheLocalAuthoritiesSuggestResultShouldBeCorrect()
     {
-        var result = _api[QueryRequestKey].Response;
+        var response = _api[SuggestRequestKey].Response;
+
+        response.Should().NotBeNull();
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
 
+        var content = await response.Content.ReadAsByteArrayAsync();
+        var results = content.FromJson<SuggestResponse<LocalAuthority>>().Results;
+        var result = results.FirstOrDefault();
         result.Should().NotBeNull();
-        result.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        result?.Text.Should().Be("*201*");
+        result?.Document?.Name.Should().Be("City of London");
+        result?.Document?.Code.Should().Be("201");
+    }
+
+    [Then("the local authorities suggest result should be:")]
+    private async Task ThenTheLocalAuthoritiesSuggestResultShouldBe(Table table)
+    {
+        var response = _api[SuggestRequestKey].Response;
+
+        response.Should().NotBeNull();
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var content = await response.Content.ReadAsByteArrayAsync();
+        var results = content.FromJson<SuggestResponse<LocalAuthority>>().Results;
+
+        var set = new List<dynamic>();
+
+        foreach (var result in results)
+        {
+            set.Add(new { result.Text, result.Document?.Name, result.Document?.Code });
+        }
+
+        table.CompareToDynamicSet(set, false);
```

Comment:
`CompareToSet()` would work better here. See `InsightCensus` feature for some examples.